### PR TITLE
feat: Generalised Custom Input

### PR DESCRIPTION
**Pull request by shivani170**
Generalized custom input

### CODE REVIEW

File: src/Common/CustomInput/CustomInput.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        if (!Array.isArray(error)) {
+            return [error]
+        }
+    
+        return error
+    }
+
+    return (
+        <div className="flex column left top">
+            {label ? (
```
Comment (vivek-devtron) [inline src/Common/CustomInput/CustomInput.tsx:61]: This should be label &&

File: src/Common/CustomInput/CustomInput.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        return (
+            <span>
+                <a target="_blank" href={link} className="cursor fs-13 onlink">
+                    {linkText}
+                </a>
+            </span>
+        )
+    }
+
+    onBlur = (e) => {
```
Comment (vivek-devtron) [inline src/Common/CustomInput/CustomInput.tsx:39]: We can directly use onBlur from props

File: src/Common/CustomInput/CustomInput.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+       onBlur && onBlur(e)
+    }
+
+    onFocus = (e) => {
```
Comment (vivek-devtron) [inline src/Common/CustomInput/CustomInput.tsx:43]: Same here

File: src/Common/CustomInput/CustomInput.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+                tabIndex={tabIndex}
+                autoFocus={autoFocus}
+            />
+            {handleError(error).map((err) => (
+                <div className="form__error" key={err}>
+                    <FormError className="form__icon form__icon--error" />
+                    {err}
+                </div>
+            ))}
+             {helperText ? (
```
Comment (vivek-devtron) [inline src/Common/CustomInput/CustomInput.tsx:92]: we can write this as `helperText && (`

File: src/Common/CustomInput/Types.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+export interface CustomInputProps {
+    name: string
+    value: string | number
+    autoComplete: string
```
Comment (vivek-devtron) [inline src/Common/CustomInput/Types.ts:4]: Why is this mandatory?

**Review by vivek-devtron — APPROVED** [review]

File: src/Common/CustomInput/Types.ts (hunk 5 above)
Comment (shivani170) [reply src/Common/CustomInput/Types.ts]: It shouldn't be, Removed!

File: src/Common/CustomInput/CustomInput.tsx (hunk 4 above)
Comment (shivani170) [reply src/Common/CustomInput/CustomInput.tsx]: Done

**Review by vivek-devtron — APPROVED** [review]

File: src/Common/CustomInput/CustomInput.tsx [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+    ref,
+    onKeyPress,
+    defaultValue,
+    onKeyDown,
+    required,
+    additionalErrorInfo
+}: CustomInputProps) {
+    const renderLabelHelperText = () => {
+        return (
+            <span>
```
Comment (vivek-devtron) [inline src/Common/CustomInput/CustomInput.tsx:38]: Why this span is requiredd?

File: src/Common/CustomInput/CustomInput.tsx [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+import { ReactComponent as Info } from '../../Assets/Icon/ic-info-filled.svg'
+import { ReactComponent as FormError } from '../../Assets/Icon/ic-warning.svg'
+
+export function CustomInput({
+    name,
+    value,
+    error,
+    onChange,
+    onFocus = (e) => {},
+    label = '',
```
Comment (vivek-devtron) [inline src/Common/CustomInput/CustomInput.tsx:11]: Why default value is '' if type is ReactNode?

File: src/Common/CustomInput/CustomInput.tsx [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+        return (
+            <div className="form__error" key={error}>
+                <FormError className="form__icon form__icon--error" />
+                {error}
+                {typeof additionalErrorInfo === 'function' && additionalErrorInfo()}
+            </div>
+        )
+    }
+
+    const getInputError = () => {
```
Comment (vivek-devtron) [inline src/Common/CustomInput/CustomInput.tsx:80]: This should return null if no error

File: src/Common/CustomInput/CustomInput.tsx [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+        }
+        if (typeof handleOnBlur === 'function') {
+            handleOnBlur(event)
+        }
+    }
+
+    const renderInputErrorMessage = (error: string) => {
+        return (
+            <div className="form__error" key={error}>
+                <FormError className="form__icon form__icon--error" />
```
Comment (vivek-devtron) [inline src/Common/CustomInput/CustomInput.tsx:73]: Rename this to FormErrorIcon or ErrorIcon

File: src/Common/CustomInput/CustomInput.tsx [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+    const getInputError = () => {
+        if (error?.length > 0) {
+            if (typeof error === 'object') {
+                return handleError(error).map((err: string) => renderInputErrorMessage(err))
+            }
+            return renderInputErrorMessage(error)
+        }
+    }
+
+    const renderInputLabel = () => {
```
Comment (vivek-devtron) [inline src/Common/CustomInput/CustomInput.tsx:89]: Return null if no label

File: package.json [hunk 11]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "@devtron-labs/devtron-fe-common-lib",
-    "version": "0.0.45",
+    "version": "0.0.46",
```
Comment (vivek-devtron) [inline package.json:3]: this should be the beta version instead 46 till the testing

File: src/Common/CustomInput/CustomInput.tsx (hunk 6 above)
Comment (shivani170) [reply src/Common/CustomInput/CustomInput.tsx]: Not required! removed

**Review by vivek-devtron — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**